lib/announcer.js

```diff
@@ -6,18 +6,20 @@ const Persistent = require('./persistent')
 const { COMMANDS } = require('./constants')
 
 module.exports = class Announcer {
-  constructor (dht, keyPair, target) {
+  constructor (dht, keyPair, target, record, opts = {}) {
     this.dht = dht
     this.keyPair = keyPair
     this.target = target
     this.relays = []
     this.stopped = false
-    this.record = c.encode(m.peer, { publicKey: this.keyPair.publicKey, relayAddresses: [] })
+    this.record = record
```

Comment:
is this change still needed now? otherwise let's move it to another pr or just keep it as it was